ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/snippet/factory/BasicSnippetFactory.java

```diff
@@ -196,4 +207,30 @@ public ExpressionSnippet createExpressionSnippet(Node node) {
         }
         return null;
     }
+
+    /**
+     * Check the input node contains isolated keyword.
+     *
+     * @param node input Node.
+     * @return node contains isolated keyword or not.
+     */
+    private boolean isContainsIsolated(Node node) {
+        if (node instanceof ModuleVariableDeclarationNode) {
+            NodeList<Token> nodeList = ((ModuleVariableDeclarationNode) node).qualifiers();
+            for (Token token: nodeList) {
+                if (token.kind().stringValue().equals("isolated")) {
+                    return true;
+                }
+            }
+        } else if (node instanceof FunctionDefinitionNode) {
+            NodeList<Token> nodeList = ((FunctionDefinitionNode) node).qualifierList();
+            for (Token token: nodeList) {
+                if (token.kind().stringValue().equals("isolated")) {
+                    return true;
+                }
+            }
```

Comment:
I see that this code segment is being duplicated. Let's get that out of the if clause and handle it. Also, instead of the for-loop approach we can use stream API with a contains check/ `anyMatch` API. That would be convenient.

Reply:
Updated in 3a77f2123f76ad957c11b56f10d1efecda8bca08